File: client/lib/model/player.dart
class Player {
  String _race;
  int _tacticPool;
  int _fleetPool;
  int _strategyPool;
  int _victoryPoints;
  int _strategyCardId;
  bool _strategyExhausted;
  bool _passed;


  Player(
    this._race,
    this._strategyExhausted,
    this._tacticPool,
    this._fleetPool,
    this._strategyPool,
    this._victoryPoints,
    this._passed,
    this._strategyCardId
  );

  String getName() {
    return _race;
  }

  void setName(String newRace) {
    _race = newRace;
  }

  int getTacticPool() {
    return _tacticPool;
  }

  int getFleetPool() {
    return _fleetPool;
  }

  int getStrategyPool() {
    return _strategyPool;
  }

  int getVictoryPoints() {
    return _victoryPoints;
  }

  int getStrategyCard() {
    return _strategyCardId;
  }

  factory Player.fromJson(Map<String, dynamic> json) {
    return switch (json) {
      {
        'race': String race,
        'strategyExhausted': bool strategyExhausted,
        'tactic': int tactic,
        'fleet': int fleet,
        'strategyPool': int strategyPool,
        'victoryPoints': int victoryPoints,
        'passed': bool passed
      } =>
        Player(
          race,
          strategyExhausted,
          tactic,
          fleet,
          strategyPool,
          victoryPoints,
          passed,
          0
        ),
      _ => throw const FormatException('Failed to load Player.'),
    };
  }

  Map<String, dynamic> toJson() => {
    "race": _race,
    "strategyExhaused": _strategyExhausted,
    "tactic": _tacticPool,
    "fleet": _fleetPool,
    "strategyPool": _strategyPool,
    "victoryPoints": _victoryPoints,
    "passed": _passed,
    "strategyCardId": _strategyCardId
  };
}

File: server/src/com/twilightimperium/backend/model/game/entities/Player.java
package com.twilightimperium.backend.model.game.entities;

public class Player {
    private String race;
    private Integer strategyCardId;
    private boolean strategyExhausted; //stores if a  player has used their strategy
    private int tactic;
    private int fleet;
    private int strategyPool;
    private int victoryPoints;
    private boolean passed;


    public Player(){
        race = "not_chosen";
        passed = false;
    }


    public String getRace() {
        return this.race;
    }

    public void setRace(String race) {
        this.race = race;
    }

    public Integer getStrategyCardId() {
        return this.strategyCardId;
    }

    public void setStrategyCardId(Integer strategyCardId) {
        this.strategyCardId = strategyCardId;
    }

    public boolean isStrategyExhausted() {
        return this.strategyExhausted;
    }

    public boolean getStrategyExhaused() {
        return this.strategyExhausted;
    }

    public void setStrategyExhausted(boolean strategyExhaused) {
        this.strategyExhausted = strategyExhaused;
    }

    public int getTactic() {
        return this.tactic;
    }

    public void setTactic(int tactic) {
        this.tactic = tactic;
    }

    public int getFleet() {
        return this.fleet;
    }

    public void setFleet(int fleet) {
        this.fleet = fleet;
    }

    public int getStrategyPool() {
        return this.strategyPool;
    }

    public void setStrategyPool(int strategyPool) {
        this.strategyPool = strategyPool;
    }

    public int getVictoryPoints() {
        return this.victoryPoints;
    }

    public void setVictoryPoints(int victoryPoints) {
        this.victoryPoints = victoryPoints;
    }

    public boolean isPassed() {
        return this.passed;
    }

    public boolean getPassed() {
        return this.passed;
    }

    public void setPassed(boolean passed) {
        this.passed = passed;
    }

    
    // Other methods
}
